sys/net/gnrc/pktbuf_static/gnrc_pktbuf_static.c

```diff
@@ -424,7 +428,9 @@ static void *_pktbuf_alloc(size_t size)
         }
     }
     else {
-        _unused_t *new = (_unused_t *)(((uint8_t *)ptr) + size);
+        /* alignment is ensured by rounding size up in the _align() function.
+         * We cast to uintptr_t as intermediate step to silence -Wcast-align */
+        _unused_t *new = (_unused_t *)((uintptr_t)ptr + size);
```

Comment:
Doesn't this result in an `size * sizeof(uintptr_t)` bytes offset? I assume `uintptr_t` to be word length.

Reply:
`uintptr_t` is an pointer sized integer. So you have integer arithmetic here, not pointer arithmetic.